Build failure on Arch/Manjaro when using mixed Qt5/Qt6 environment

Hi,
When trying to build ntfydesktop on manjaro, the build fails with

```
=> Beginne build()...
-- The C compiler identification is GNU 15.1.1
-- The CXX compiler identification is GNU 15.1.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Installing in the same prefix as Qt, adopting their path scheme.
-- CPM: Adding package json@3.12.0 (v3.12.0)
-- Using the multi-header code from /home/tboy/.cache/yay/ntfydesktop/src/build/_deps/json-src/include/
-- CPM: Adding package emojicpp@2.0.1 (v2.0.1)
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Found CURL: /usr/lib/libcurl.so (found version "8.15.0")
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found OpenGL: /usr/lib/libOpenGL.so
-- Found WrapOpenGL: TRUE
-- Found WrapVulkanHeaders: /usr/include
-- Found Gettext: /usr/bin/msgmerge (found version "0.25.1")
CMake Error at /usr/lib/cmake/Qt6Qml/Qt6QmlMacros.cmake:3651 (message):
qt_generate_foreign_qml_types() is only available in Qt 6.
Call Stack (most recent call first):
/usr/lib/cmake/Qt6Qml/Qt6QmlConfig.cmake:210 (include)
/usr/share/cmake/Modules/CMakeFindDependencyMacro.cmake:78 (find_package)
/usr/lib/cmake/KF6Config/KF6ConfigConfig.cmake:52 (find_dependency)
/usr/share/cmake/Modules/CMakeFindDependencyMacro.cmake:78 (find_package)
/usr/lib/cmake/KF6XmlGui/KF6XmlGuiConfig.cmake:44 (find_dependency)
CMakeLists.txt:58 (find_package)


-- Configuring incomplete, errors occurred!
```

To fix this, please add "-DQT_DEFAULT_MAJOR_VERSION=6" to the cmake options:

```
--- /home/box/.cache/yay/ntfydesktop/PKGBUILD	2025-08-09 01:07:13.472264995 +0200
+++ /tmp/ntfydsk/PKGBUILD	2025-08-09 10:02:48.265866891 +0200
@@ -19,6 +19,7 @@
     -W no-dev
     -D CMAKE_BUILD_TYPE=Release
     -D CMAKE_INSTALL_PREFIX=/usr
+    -DQT_DEFAULT_MAJOR_VERSION=6
   )
   cmake "${cmake_options[@]}"
   cmake --build build
```


Keep up the good work :)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,6 +22,7 @@ set(GEN_DIR "${CMAKE_BINARY_DIR}/generated")
 include_directories("${GEN_DIR}")
 
 # Libraries
+set(QT_DEFAULT_MAJOR_VERSION "6")
 find_package(ECM REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
 include(KDEInstallDirs) # This is supposed to be KDEInstallDirs6 but KDEInstallDirs6 is broken atm

--- a/scripts/arch/PKGBUILD
+++ b/scripts/arch/PKGBUILD
@@ -19,6 +19,7 @@ build() {
     -W no-dev
     -D CMAKE_BUILD_TYPE=Release
     -D CMAKE_INSTALL_PREFIX=/usr
+    -D QT_DEFAULT_MAJOR_VERSION=6 # TODO: Remove after v1.5.1, will be fixed in CMakeLists.txt
   )
   cmake "${cmake_options[@]}"
   cmake --build build

